pkg/api/component_report.go

```diff
@@ -0,0 +1,601 @@
+			Name:  "Network",
+			Value: c.Network,
+		})
+	}
+	if c.Platform != "" {
+		queryString += ` AND platform = @Platform`
+		commonParams = append(commonParams, bigquery.QueryParameter{
+			Name:  "Platform",
+			Value: c.Platform,
+		})
+	}
+	if c.TestID != "" {
+		queryString += ` AND test_id = @TestId`
+		commonParams = append(commonParams, bigquery.QueryParameter{
+			Name:  "TestId",
+			Value: c.TestID,
+		})
+	}
+
+	if c.ExcludePlatforms != "" {
+		queryString += ` AND platform NOT IN UNNEST(@ExcludePlatforms)`
+		commonParams = append(commonParams, bigquery.QueryParameter{
+			Name:  "ExcludePlatforms",
+			Value: strings.Split(c.ExcludePlatforms, ","),
+		})
+	}
+	if c.ExcludeArches != "" {
+		queryString += ` AND arch NOT IN UNNEST(@ExcludeArches)`
+		commonParams = append(commonParams, bigquery.QueryParameter{
+			Name:  "ExcludeArches",
+			Value: strings.Split(c.ExcludeArches, ","),
+		})
+	}
+	if c.ExcludeNetworks != "" {
+		queryString += ` AND network NOT IN UNNEST(@ExcludeNetworks)`
+		commonParams = append(commonParams, bigquery.QueryParameter{
+			Name:  "ExcludeNetworks",
+			Value: strings.Split(c.ExcludeNetworks, ","),
+		})
+	}
+	if c.ExcludeUpgrades != "" {
+		queryString += ` AND upgrade NOT IN UNNEST(@ExcludeUpgrades)`
+		commonParams = append(commonParams, bigquery.QueryParameter{
+			Name:  "ExcludeUpgrades",
+			Value: strings.Split(c.ExcludeUpgrades, ","),
+		})
+	}
+	if c.ExcludeVariants != "" {
+		queryString += ` AND variant NOT IN UNNEST(@ExcludeVariants)`
+		commonParams = append(commonParams, bigquery.QueryParameter{
+			Name:  "ExcludeVariants",
+			Value: strings.Split(c.ExcludeVariants, ","),
+		})
+	}
+
+	groupString := `
+		GROUP BY
+		network,
+		upgrade,
+		arch,
+		platform,
+		test_id `
+
+	baseString := queryString + ` AND branch = @BaseRelease`
+	baseQuery := c.client.Query(baseString + groupString)
+	baseQuery.Parameters = append(commonParams, []bigquery.QueryParameter{
+		{
+			Name:  "From",
+			Value: c.baseRelease.Start,
+		},
+		{
+			Name:  "To",
+			Value: c.baseRelease.End,
+		},
+		{
+			Name:  "BaseRelease",
+			Value: c.baseRelease.Release,
+		},
+	}...)
+
+	var baseStatus, sampleStatus map[apitype.ComponentTestIdentification]apitype.ComponentTestStats
+	var baseErrs, sampleErrs []error
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		baseStatus, baseErrs = c.fetchTestStatus(baseQuery)
+	}()
+
+	sampleString := queryString + ` AND branch = @SampleRelease`
+	sampleQuery := c.client.Query(sampleString + groupString)
+	sampleQuery.Parameters = append(commonParams, []bigquery.QueryParameter{
+		{
+			Name:  "From",
+			Value: c.sampleRelease.Start,
+		},
+		{
+			Name:  "To",
+			Value: c.sampleRelease.End,
+		},
+		{
+			Name:  "SampleRelease",
+			Value: c.sampleRelease.Release,
+		},
+	}...)
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		sampleStatus, sampleErrs = c.fetchTestStatus(sampleQuery)
+	}()
+	wg.Wait()
```

Comment:
The queries are slow so the UI gives the user the ability to cancel.  However, the UI has no mechanism to cancel the api call already "in flight" so users can then make new queries and the existing ones keep running.

We can either add an ability to cancel in flight api calls or disable the Cancel button on the UI for now.